uploadBuildToS3.js

```diff
@@ -0,0 +1,182 @@
+'use strict'
+/**
+ * This script used to upload js, css from dist folder to S3
+ *
+ * node uploadBuildToS3.js
+ *
+ * To copy dist folder to Yelda repo include the yelda repo static folder path in the command
+ * Like this
+ * npm run uploadBuildToS3 '/home/Documents/www/yelda/git/yelda/frontend/static'
+ * This command uploadBuildToS3 runs build, test, upload to s3, npm version update, push to git and move dist folder to yelda repo
+ */
+
+require('dotenv').config({
+  path: process.env.DOTENV || '.env'
+})
+
+const AWS = require('aws-sdk')
+const fs = require('fs')
+const path = require('path')
+const mime = require('mime')
+const { exec } = require('child_process')
+
+const main = async () => {
+  try {
+    const branchName = await getCurrentBranch()
+
+    const version = await updateNPMVersion()
+    console.info(version)
+
+    const publishResponse = await publishPackage()
+    console.info(publishResponse)
+
+    await pushToGit(branchName)
+    await uploadToS3()
+    await copyDistToYelda()
+
+    process.exit()
+  } catch (err) {
+    console.error('Error while uploading to s3:', err)
+    process.exit()
+  }
+}
+
+/**
+ * Executes the given command
+ * @param {String} cmd
+ * @return {Promise<string>} - stdout
+ */
+const executeCommand = (cmd) => {
+  return new Promise((resolve, reject) => {
+    exec(cmd, (error, stdout) => {
+      if (error) {
+        reject(error)
+      }
+
+      resolve(typeof stdout === 'string' ? stdout.trim() : false)
+    })
+  })
+}
+
+/**
+ * Get the current git branch name
+ * @return {Promise<string>} stdout
+ */
+const getCurrentBranch = async() => {
+  return await executeCommand('git rev-parse --abbrev-ref HEAD')
+}
+
+/**
+ * Updates the NPM version & git commit added with the updated version
+ *  @return {Promise<string>} stdout
+ */
+const updateNPMVersion = async () => {
+  console.info('Updating npm version ...')
+  return await executeCommand('npm -no-git-tag-version version patch')
```

Comment:
Question here
The current version is 1.0.49
Using this command, the version will be updated to 1.0.50 right ?
But what if we want to update to 1.1.0 or even to 2.0.0 ?

I think we should add an option to the script comment to choose the update type ? This way we could dynamically execute the command:
`npm -no-git-tag-version version patch` OR `npm -no-git-tag-version version minor` OR `npm -no-git-tag-version version major` ?
But we could keep `patch` as default if no update type as been precised.

What do you think ? @meelie

Reply:
Good feedback.
Can be done when we will need it ;)